flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapRestoreOperation.java

```diff
@@ -201,11 +201,13 @@ private void readStateHandleStateData(
             long offset = groupOffset.f1;
 
             // Check that restored key groups all belong to the backend.
-            Preconditions.checkState(
-                    keyGroupRange.contains(keyGroupIndex),
-                    "Key group %s doesn't belong to this backend with key group range: %s",
-                    keyGroupIndex,
-                    keyGroupRange);
+            if (!keyGroupRange.contains(keyGroupIndex)) {
```

Comment:
Currently, `HeapRestoreOperation.java` is both used for checkpoint and savepoint restore as they actually have the same format. Thus, shall we throw exception once this key group does not belong to current key group range during savepoint restore?